CostumeStoreDB/1 table.sql

```diff
@@ -0,0 +1,30 @@
+use CostumeStoreDB
+go
+
+drop table if exists Costume
+go
+
+create table dbo.Costume(
+    CostumeId int not null identity primary key,
+    CustomerFirstName varchar(20) not null constraint ck_Costume_CustomerFirstName_cannot_be_blank check(CustomerFirstName <> ''),
+    CustomerLastName varchar(20) not null constraint ck_Costume_CustomerLastName_cannot_be_blank check(CustomerLastName <> ''),
+    Style varchar(20) not null constraint ck_Costume_Style_must_be_one_of_the_style_options check(Style in('American Girl Doll', 'Artist', 'Bumble Bee', 'Colonial Boy', 'Colonial Girl', 'Elephant', 'Fire Man', 'Police Man', 'Princess', 'Zebra')),
+    Size varchar(2) not null constraint ck_Costume_Size_must_be_one_of_the_size_options check(Size in ('XS', 'S', 'M', 'L', 'XL')),
+    SoldQuantity int not null constraint ck_Costume_SoldQuantity_must_be_greater_than_zero check(SoldQuantity > 0),
+    PurchaseDate date not null constraint ck_Costume_PurchaseDate_must_be_after_Jan_1_2020 check(datefromparts(2020, 1, 1) < PurchaseDate),
+    SoldDate date not null,
+    PurchasePrice as case Size when 'XS' then 15.00 when 'S' then 17.00 when 'M' then 20.00 when 'L' then 22.00 when 'XL' then 25.00 end persisted,
+    --BG I didnt do the SoldPrice as a case on size because she may sell to a friend or family member for cost price, as per the spec
+    SoldPrice decimal(4,2) not null,
+    CustomerTotalPrice as SoldPrice * SoldQuantity persisted,
```

Comment:
For those that are paying cost price the calculation will be different.

Reply:
I'm not sure why. I think the calculation works even when they are paying cost price. SoldPrice is not a case on size it is inserted individually for each person how much s/he paid. So, the SoldPrice * SoldQuantity = CustomerSoldPrice across the board, according to my understanding....

Reply:
Yes I misunderstood.